Assets/Scripts/Robot/CommunicationManager.cs

```diff
@@ -52,13 +52,13 @@ public readonly struct SensedObject<T>
     {
         public readonly float Distance;
         public readonly float Angle;
-        public readonly T Item;
+        public readonly T RobotId;
```

Comment:
Do not call it RobotId. This thing is generic. It makes no sense.

Reply:
It is only used to store the robot id in all the used places. Therefore it should be called that instead.

Reply:
Then make it non-generic.